lms/services/rsa_key.py

```diff
@@ -115,15 +115,19 @@ def _as_pem_private_key(self, rsa_key, aes_iv) -> bytes:
         return self._aes_service.encrypt(aes_iv, pem_private_key)
 
     @staticmethod
-    def _as_jwk_public_key(rsa_key) -> dict:
-        """Encode a `jose.jwt.RSAKey` as a JWK public key."""
-        pem_public_key = rsa_key.public_key().public_bytes(
-            encoding=serialization.Encoding.PEM,
-            format=serialization.PublicFormat.SubjectPublicKeyInfo,
-        )
-        return jwk.RSAKey(  # type: ignore
-            algorithm=constants.Algorithms.RS256, key=pem_public_key.decode("utf-8")
-        ).to_dict()
+    def _as_jwk_public_key(rsa_key: rsa.RSAPrivateKey) -> dict:
+        """Encode a  as a JWK public key."""
+        public_key_numbers = rsa_key.public_key().public_numbers()
```

Comment:
All we used python-jose here was to generate this. It's just formatting these numbers so I'm conformatible doing it ourselves.